### DISCUSSION TITLE
metrics about the test

### DISCUSSION THREAD

**Discussion by RitaFerreira03** [ultralytics/yolov5#5369] [Q&A]
Hello
Is it possible to generate the metrics results that are output on train and validation, but for the test?

**Comment by glenn-jocher** [comment]
@RitaFerreira03 to evaluate a model on each dataset split:

```bash
python val.py --task train
python val.py --task val  # default
python val.py --task test
```

**Reply by RitaFerreira03** [reply]
Do you mean that after training, validating and testing (detecting) the model I can run the command of validation again with the argument "--task test"? If I do that I get an error "KeyError: 'test'" and I tried --task Test and nothing happened... 
maybe I'm not understanding something or I'm doing something wrong.

After training and testing the results I got are really good! I already have the metrics for the training and I'm just trying to output the metrics for my results.

Thank you for your reply and help!

**Reply by RitaFerreira03** [reply]
I think I found out why! I think the error was in the yaml file which didn't have the path to the test dataset. 
Can you please confirm if that makes sense? 

Thank you once again!

**Reply by glenn-jocher** [reply]
@RitaFerreira03 above commands run val.py on your 3 datasets splits. Test set is optional.
https://github.com/ultralytics/yolov5/blob/a4fece8c1480aed46a38a6344b403d79c81bd751/data/VisDrone.yaml#L12-L14

**Reply by RitaFerreira03** [reply]
yes, that is what I have in the yaml file. 
I just wanted to know if I can have metrics output when I run detect.py and how. The output are the test images with the objects detected, but that's it?

**Reply by glenn-jocher** [reply]
@RitaFerreira03 val.py produces metrics. detect.py simply runs inference.

**Reply by RitaFerreira03** [reply]
Thank you @glenn-jocher. Nice work with the algorithm!
